### DISCUSSION TITLE
Starting a transaction in a task is not possible

### DISCUSSION THREAD

**Discussion by buseynehannes** [Snowflake-Labs/schemachange#374] [Q&A]
**Describe the bug**
I'm trying to start a transaction in a task to execute multiple statements

**To Reproduce**
1. Create an example migration with the following task
```sql
CREATE OR REPLACE TASK EXAMPLE_TASK
    WAREHOUSE = EXPLORATION
    SCHEDULE = '5 minutes'
AS
    BEGIN
    START TRANSACTION;
    SELECT * FROM SOME_TABLE;
    SELECT * FROM SOME_TABLE;
    COMMIT;
END;
```
2. execute the migration -> fails on EOF error
3. render and execute in snowflake worksheet -> works perfectly

**Expected behavior**
Since the rendered template works in a snowflake worksheet, I'd expect this to work in a migration too

**Schemachange (please complete the following information):**
- Version (e.g. 3.5.3): 3.6.1

**Additional context**
I'm using [Repeatable read isolation](https://docs.snowflake.com/en/user-guide/streams-intro#repeatable-read-isolation) in my task, so I need the transaction, tried using `BEGIN TRANSACTION` too, same issue.

**Comment by sfc-gh-tmathew** [comment]
Hey @buseynehannes,

The issue is that execute_string() splits statements on semicolons, which breaks your BEGIN...END block into separate statements.

Solution: Use $$ delimiters to protect the block:

```sql
CREATE OR REPLACE TASK EXAMPLE_TASK
WAREHOUSE = EXPLORATION
SCHEDULE = '5 minutes'
AS
EXECUTE IMMEDIATE
$$
BEGIN    
    START TRANSACTION;
    SELECT * FROM SOME_TABLE;
    SELECT * FROM SOME_TABLE;
    COMMIT;
END;
$$;
```

This tells execute_string() to treat everything between $$ as a single statement, preventing it from splitting on the internal semicolons.

Reference: [Snowflake CREATE TASK documentation](https://docs.snowflake.com/en/sql-reference/sql/create-task) supports both single-quoted and dollar-quoted ($$) string syntax.

This should work through schemachange without needing a stored procedure workaround! Otherwise, you could add the multiple statements to a stored proc and then call the stored proc in the task.

**Reply by sfc-gh-tmathew** [reply]
Similar question came up in https://github.com/Snowflake-Labs/schemachange/discussions/389
